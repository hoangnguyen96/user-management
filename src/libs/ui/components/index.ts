// Common
export * from "./Common";

// Components
export { default as HeadingContent } from "./HeadingContent";
export { default as LazyFallbackContent } from "./LazyFallbackContent";
export { default as Logo } from "./Logo";
export { default as Footer } from "./Footer";
export { default as FormController } from "./FormController";
export { default as ListCustomersContent } from "./ListCustomersContent";
export { default as ListProductContent } from "./ListProductContent";
export { default as NavigationBar } from "./NavigationBar";
export { default as Temporary } from "./Temporary";
export { default as TitleContent } from "./TitleContent";
export { default as MenuBar } from "./MenuBar";
export { default as ErrorBoundary } from "./ErrorBoundary";
export { default as ModalCreateUserSuccess } from "./Modal/ModalCreateUserSuccess";
