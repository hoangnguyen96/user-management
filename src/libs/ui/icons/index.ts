export * from "./HelpIcon";
export * from "./IncomeIcon";
export * from "./KeyIcon";
export * from "./Members";
export * from "./ProductIcon";
export * from "./PromoteIcon";
export * from "./Screen";
export * from "./SearchIcon";
export * from "./TotalCustomers";
export * from "./UserIcon";
