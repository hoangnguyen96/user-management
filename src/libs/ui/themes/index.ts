import { createTheme } from "@mui/material/styles";
import { breakpointsBase } from "./breakpoints";
import { colors } from "./colors";
import { typographyBase } from "./typography";

export const themeDefault = createTheme({
  breakpoints: breakpointsBase,
  palette: colors,
  typography: typographyBase,

  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          textTransform: "capitalize",
        },
      },
      variants: [
        {
          props: { variant: "contained" },
          style: {
            color: "White",
          },
        },
        {
          props: { variant: "activate" },
          style: {
            backgroundColor: "rgba(22, 192, 152, 0.38)",
            color: "#008767",
            border: "1px solid #00b087",
            "&:hover": {
              backgroundColor: "rgba(22, 192, 152, 0.62)",
            },
          },
        },
        {
          props: { variant: "inactivate" },
          style: {
            backgroundColor: "#ffc5c5",
            color: "#df0404",
            border: "1px solid #df0404",
            "&:hover": {
              backgroundColor: "rgb(253 170 170)",
            },
          },
        },
        {
          props: { variant: "normal" },
          style: {
            backgroundColor: " rgba(245, 245, 245, 1)",
            color: "rgba(64, 75, 82, 1)",
            border: "1px solid #eeeeee",
            "&:hover": {
              backgroundColor: "rgb(219 219 219)",
            },
          },
        },
      ],
      defaultProps: {
        size: "medium",
      },
    },

    MuiTextField: {
      styleOverrides: {
        root: ({ theme }) => ({
          "& .MuiOutlinedInput-root": {
            borderRadius: "12px",
            fontSize: "14px",
            backgroundColor: theme.palette.common.white,
            "& .css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input": {
              padding: "12px",
            },
            "& fieldset": {
              borderColor: "#b5b7C0",
            },
            "&:hover fieldset": {
              borderColor: theme.palette.primary.main,
            },
            "&.Mui-focused fieldset": {
              borderColor: theme.palette.primary.main,
            },
          },
        }),
      },
    },

    MuiSelect: {
      styleOverrides: {
        root: {
          width: "auto",
          flex: 1,
        },
      },
    },

    MuiFormControl: {
      styleOverrides: {
        root: {
          flexDirection: "row",
          justifyContent: "center",
          alignItems: "center",
          backgroundColor: "#f9fbff",
          borderRadius: 12,
          minWidth: 154,
          "& .css-wama2x-MuiFormLabel-root-MuiInputLabel-root": {
            position: "unset",
            transform: "unset",
            left: "unset",
            top: "unset",
          },
          "& .css-cb5d54-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root":
            {
              "& fieldset": {
                border: 0,
              },
              borderColor: "transparent",
              borderWidth: 0,
            },
          "& .css-wxw8j3-MuiFormLabel-root-MuiInputLabel-root": {
            position: "unset",
            transform: "unset",
            left: "unset",
            top: "unset",
            flex: 0.8,
            padding: 10,
            paddingRight: 5,
            fontSize: "12px",
            textAlign: "right",
          },
          "& .css-zbhh1g-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input.MuiSelect-select":
            {
              padding: "11px 32px 11px 0",
              fontSize: "12px",
              fontWeight: 500,
            },
          "& .css-cb5d54-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root.Mui-focused .MuiOutlinedInput-notchedOutline":
            {
              borderColor: "transparent",
              borderWidth: 0,
            },
          "& .css-10d36jd-MuiSvgIcon-root-MuiSelect-icon, & .css-5ajf1d-MuiSvgIcon-root-MuiSelect-icon":
            {
              width: 20,
              height: 20,
            },
          "& .css-1j5jes1-MuiFormLabel-root.Mui-focused": {
            color: "rgba(0, 0, 0, 0.6)",
          },
        },
      },
    },

    MuiPaper: {
      styleOverrides: {
        root: {
          "& .css-ddqxy4-MuiButtonBase-root-MuiMenuItem-root": {
            fontSize: "12px",
          },
        },
      },
    },

    MuiListItemButton: {
      styleOverrides: {
        root: ({ theme }) => ({
          borderRadius: theme.spacing(1),
          "&.Mui-selected": {
            backgroundColor: theme.palette.primary.main,
            color: theme.palette.common.white,
          },
          "&.Mui-selected:hover": {
            backgroundColor: theme.palette.primary.main,
            color: theme.palette.common.white,
          },
        }),
      },
    },

    MuiLink: {
      styleOverrides: {
        root: ({ theme }) => ({
          backgroundColor: "transparent",
          fontSize: "14px",
          textDecoration: "none",
          color: theme.palette.primary.contrastText,
          "&:hover": {
            cursor: "pointer",
          },
        }),
      },
    },

    MuiPagination: {
      styleOverrides: {
        root: ({ theme }) => ({
          "& .MuiPagination-ul": {
            "& .MuiButtonBase-root.MuiPaginationItem-root": {
              fontSize: "12px",
              fontWeight: 500,
              backgroundColor: "#f5f5f5",
              minWidth: "25px",
              height: "24px",
              margin: "0 5px",
              "&.Mui-selected": {
                backgroundColor: theme.palette.primary.main,
                color: "white",
              },
              "& .MuiSvgIcon-root.MuiPaginationItem-icon": {
                fontSize: "12px",
              },
            },
          },
        }),
      },
    },

    MuiContainer: {
      styleOverrides: {
        root: {
          maxWidth: "100%",
          paddingLeft: 0,
          paddingRight: 0,
          "@media (min-width:600px)": {
            paddingLeft: 0,
            paddingRight: 0,
          },
          "@media (min-width:960px)": {
            paddingLeft: 0,
            paddingRight: 0,
          },
          "@media (min-width:1280px)": {
            paddingLeft: 0,
            paddingRight: 0,
          },
          "@media (min-width:1440px)": {
            maxWidth: "1440px",
          },
        },
      },
    },
  },
});
