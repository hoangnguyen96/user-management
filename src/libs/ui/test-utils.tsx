import { PropsWithChildren, ReactElement, ReactNode } from "react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { BrowserRouter } from "react-router-dom";
import { RenderOptions, render } from "@testing-library/react";

// ThemeProvider
import { ThemeProvider } from "@mui/material";

// Config
import { themeDefault } from "./themes";

const AllTheProviders = ({ children }: PropsWithChildren) => {
  return <ThemeProvider theme={themeDefault()}>{children}</ThemeProvider>;
};

export const wrapper = ({ children }: { children: ReactNode }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return (
    <ThemeProvider theme={themeDefault()}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderWithQueryClient = (
  ui: ReactElement,
  options?: RenderOptions
) => render(ui, { wrapper, ...options });

export const WrapperRouter = ({ children }: { children: ReactNode }) => (
  <BrowserRouter>{children}</BrowserRouter>
);

export { customRender as render };
export { customRenderWithQueryClient as renderWithQueryClient };
