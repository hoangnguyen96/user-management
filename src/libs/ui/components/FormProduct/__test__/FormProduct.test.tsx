import { renderWithQueryClient } from "@app/ui/test-utils";
import { fireEvent } from "@testing-library/react";

// Utils
import * as utils from "@app/utils";

// Component
import FormController from "..";

// Mock utils
jest.mock("@app/utils", () => ({
  clearErrorOnChange: jest.fn(),
  isEnableSubmitButton: jest.fn(),
  validatePrice: jest.fn().mockReturnValue(true),
  validateQuantity: jest.fn().mockReturnValue(true),
  validateRequired: jest.fn().mockReturnValue(true),
}));

describe("FormProduct component", () => {
  const mockOnSubmit = jest.fn();
  const mockOnNavigate = jest.fn();

  const props = {
    isOpen: true,
    isLoading: false,
    itemUpdate: { name: "kim" },
    onSubmit: mockOnSubmit,
    onNavigate: mockOnNavigate,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("Should render snapshot correctly", () => {
    expect(
      renderWithQueryClient(<FormController {...props} />)
    ).toMatchSnapshot();
  });

  it("Should render snapshot correctly when isAdmin is true", () => {
    expect(
      renderWithQueryClient(<FormController {...props} isAdmin={true} />)
    ).toMatchSnapshot();
  });

  it("Should prevents non-numeric characters from being entered in price and quantity fields", () => {
    const { getByPlaceholderText } = renderWithQueryClient(
      <FormController {...props} isAdmin={true} />
    );

    const priceInput = getByPlaceholderText("100");
    const quantityInput = getByPlaceholderText("1000");

    const nonNumericEvent = new KeyboardEvent("keydown", {
      key: "a",
      bubbles: true,
      cancelable: true,
    });
    const preventDefaultSpy = jest.spyOn(nonNumericEvent, "preventDefault");

    fireEvent(priceInput, nonNumericEvent);
    fireEvent(quantityInput, nonNumericEvent);

    expect(preventDefaultSpy).toHaveBeenCalledTimes(2);
  });

  it("Should prevents input if length exceeds 10 characters except for Backspace and Delete", () => {
    const { getByPlaceholderText } = renderWithQueryClient(
      <FormController {...props} isAdmin={true} itemUpdate={{}} />
    );

    const priceInput = getByPlaceholderText("100");

    // Simulate typing 10 characters
    fireEvent.change(priceInput, { target: { value: "12345678901" } });
    expect(priceInput).toHaveValue("12345678901");

    // Simulate typing another character, which should be prevented
    const longInputEvent = new KeyboardEvent("keydown", {
      key: "1",
      bubbles: true,
      cancelable: true,
    });
    const preventDefaultSpy = jest.spyOn(longInputEvent, "preventDefault");
    fireEvent(priceInput, longInputEvent);

    // Simulate pressing Backspace, which should not be prevented
    const backspaceEvent = new KeyboardEvent("keydown", {
      key: "Backspace",
      bubbles: true,
      cancelable: true,
    });
    preventDefaultSpy.mockClear();
    fireEvent(priceInput, backspaceEvent);
    expect(preventDefaultSpy).not.toHaveBeenCalled();

    // Simulate pressing Delete, which should not be prevented
    const deleteEvent = new KeyboardEvent("keydown", {
      key: "Delete",
      bubbles: true,
      cancelable: true,
    });
    preventDefaultSpy.mockClear();
    fireEvent(priceInput, deleteEvent);
    expect(preventDefaultSpy).not.toHaveBeenCalled();
  });

  it("Should calls onSubmit handler on form submit", () => {
    jest.spyOn(utils, "isEnableSubmitButton").mockReturnValue(true);

    const { getByPlaceholderText, getByTestId, getByText } =
      renderWithQueryClient(<FormController {...props} />);

    const nameProductInput = getByPlaceholderText("name");
    fireEvent.change(nameProductInput, { target: { value: "Banana" } });

    fireEvent.click(getByTestId("button-product"));

    expect(getByText("Update")).toBeInTheDocument();
  });

  it("Should clears errors when changing input values", () => {
    jest.spyOn(utils, "isEnableSubmitButton").mockReturnValue(true);
    const mockedClearErrorOnChange =
      utils.clearErrorOnChange as jest.MockedFunction<
        typeof utils.clearErrorOnChange
      >;

    const { getByPlaceholderText } = renderWithQueryClient(
      <FormController {...props} />
    );

    const fullNameInput = getByPlaceholderText("name");
    fireEvent.change(fullNameInput, { target: { value: "banana" } });

    expect(mockedClearErrorOnChange).toHaveBeenCalledWith(
      "name",
      {},
      expect.any(Function)
    );
  });
});
