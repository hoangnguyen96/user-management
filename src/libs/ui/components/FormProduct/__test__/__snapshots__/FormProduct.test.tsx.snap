// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormProduct component Should render snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root css-e3ss4v-MuiFormControl-root"
      >
        <div
          class="MuiStack-root css-1l4xmop-MuiStack-root"
        >
          <div
            class="MuiStack-root css-1lv7di1-MuiStack-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1j5jes1-MuiFormLabel-root"
            >
              Name:
            </label>
            <div
              class="MuiStack-root css-aubdti-MuiStack-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-mzr8ih-MuiFormControl-root-MuiTextField-root"
                data-testid="name-product"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t64y2h-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r0:"
                    name="name"
                    placeholder="name"
                    type="text"
                    value="kim"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiStack-root css-1lv7di1-MuiStack-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1j5jes1-MuiFormLabel-root"
            >
              Quantity:
            </label>
            <div
              class="MuiStack-root css-aubdti-MuiStack-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-mzr8ih-MuiFormControl-root-MuiTextField-root"
                data-testid="quantity-product"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t64y2h-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r1:"
                    inputmode="numeric"
                    name="quantity"
                    placeholder="1000"
                    type="text"
                    value="0"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1bhl3b2-MuiButtonBase-root-MuiButton-root"
          data-testid="button-product"
          disabled=""
          tabindex="-1"
          type="button"
        >
          Update
          <div
            class="MuiBox-root css-7u9dva"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root css-e3ss4v-MuiFormControl-root"
    >
      <div
        class="MuiStack-root css-1l4xmop-MuiStack-root"
      >
        <div
          class="MuiStack-root css-1lv7di1-MuiStack-root"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1j5jes1-MuiFormLabel-root"
          >
            Name:
          </label>
          <div
            class="MuiStack-root css-aubdti-MuiStack-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-mzr8ih-MuiFormControl-root-MuiTextField-root"
              data-testid="name-product"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t64y2h-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  name="name"
                  placeholder="name"
                  type="text"
                  value="kim"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiStack-root css-1lv7di1-MuiStack-root"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1j5jes1-MuiFormLabel-root"
          >
            Quantity:
          </label>
          <div
            class="MuiStack-root css-aubdti-MuiStack-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-mzr8ih-MuiFormControl-root-MuiTextField-root"
              data-testid="quantity-product"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t64y2h-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r1:"
                  inputmode="numeric"
                  name="quantity"
                  placeholder="1000"
                  type="text"
                  value="0"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1bhl3b2-MuiButtonBase-root-MuiButton-root"
        data-testid="button-product"
        disabled=""
        tabindex="-1"
        type="button"
      >
        Update
        <div
          class="MuiBox-root css-7u9dva"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FormProduct component Should render snapshot correctly when isAdmin is true 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root css-e3ss4v-MuiFormControl-root"
      >
        <div
          class="MuiStack-root css-1l4xmop-MuiStack-root"
        >
          <div
            class="MuiStack-root css-1lv7di1-MuiStack-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1j5jes1-MuiFormLabel-root"
            >
              Name:
            </label>
            <div
              class="MuiStack-root css-aubdti-MuiStack-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-mzr8ih-MuiFormControl-root-MuiTextField-root"
                data-testid="name-product"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t64y2h-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r2:"
                    name="name"
                    placeholder="name"
                    type="text"
                    value="kim"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiStack-root css-1lv7di1-MuiStack-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1j5jes1-MuiFormLabel-root"
            >
              Price:
            </label>
            <div
              class="MuiStack-root css-aubdti-MuiStack-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-mzr8ih-MuiFormControl-root-MuiTextField-root"
                data-testid="price-product"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t64y2h-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r3:"
                    inputmode="numeric"
                    name="price"
                    placeholder="100"
                    type="text"
                    value="0"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiStack-root css-1lv7di1-MuiStack-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1j5jes1-MuiFormLabel-root"
            >
              Quantity:
            </label>
            <div
              class="MuiStack-root css-aubdti-MuiStack-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-mzr8ih-MuiFormControl-root-MuiTextField-root"
                data-testid="quantity-product"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t64y2h-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r4:"
                    inputmode="numeric"
                    name="quantity"
                    placeholder="1000"
                    type="text"
                    value="0"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1bhl3b2-MuiButtonBase-root-MuiButton-root"
          data-testid="button-product"
          disabled=""
          tabindex="-1"
          type="button"
        >
          Update
          <div
            class="MuiBox-root css-7u9dva"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root css-e3ss4v-MuiFormControl-root"
    >
      <div
        class="MuiStack-root css-1l4xmop-MuiStack-root"
      >
        <div
          class="MuiStack-root css-1lv7di1-MuiStack-root"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1j5jes1-MuiFormLabel-root"
          >
            Name:
          </label>
          <div
            class="MuiStack-root css-aubdti-MuiStack-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-mzr8ih-MuiFormControl-root-MuiTextField-root"
              data-testid="name-product"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t64y2h-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r2:"
                  name="name"
                  placeholder="name"
                  type="text"
                  value="kim"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiStack-root css-1lv7di1-MuiStack-root"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1j5jes1-MuiFormLabel-root"
          >
            Price:
          </label>
          <div
            class="MuiStack-root css-aubdti-MuiStack-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-mzr8ih-MuiFormControl-root-MuiTextField-root"
              data-testid="price-product"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t64y2h-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r3:"
                  inputmode="numeric"
                  name="price"
                  placeholder="100"
                  type="text"
                  value="0"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiStack-root css-1lv7di1-MuiStack-root"
        >
          <label
            class="MuiFormLabel-root MuiFormLabel-colorPrimary css-1j5jes1-MuiFormLabel-root"
          >
            Quantity:
          </label>
          <div
            class="MuiStack-root css-aubdti-MuiStack-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-mzr8ih-MuiFormControl-root-MuiTextField-root"
              data-testid="quantity-product"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1t64y2h-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r4:"
                  inputmode="numeric"
                  name="quantity"
                  placeholder="1000"
                  type="text"
                  value="0"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1bhl3b2-MuiButtonBase-root-MuiButton-root"
        data-testid="button-product"
        disabled=""
        tabindex="-1"
        type="button"
      >
        Update
        <div
          class="MuiBox-root css-7u9dva"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
